### PR TITLE
Forking beegfs-client service which persists on reboot

### PR DESCRIPTION
**Pull request by brtknr**

### CODE REVIEW

File: tasks/client.yml [hunk 1]
```diff
@@ -96,4 +96,28 @@
...
   when: client_mgmt_host == inventory_hostname
   become: true
+
+- name: Copy remote beegfs-client.service to /etc/systemd/system/
+  copy:
+    remote_src: true
+    src: /usr/lib/systemd/system/beegfs-client.service
+    dest: /etc/systemd/system/beegfs-client.service
+  notify: Restart BeeGFS client service
+  become: true
```
Comment (markgoddard) [inline tasks/client.yml]: I don't think you need this copy - any config in /etc/ should override the defaults in /usr.

Comment (brtknr) [reply tasks/client.yml]: The trouble is there is no equivalent beegfs-client.service under /etc/systemd/system at the moment...

Comment (dougszumski) [reply tasks/client.yml]: Could you just copy a template over to /etc/.. for the bits you need to customise? I did something not too dissimilar here: https://github.com/stackhpc/ansible-role-os-monasca-monitor/blob/master/templates/monasca-monitor.target.j2

Comment (brtknr) [reply tasks/client.yml]: Thanks for spelling that out... did realise you could do partial override of unit file config like that... I've done what you've asked for but I'm getting this:
```
beegfs-client.service lacks both ExecStart= and ExecStop= setting. Refusing.
```
Do you need to supply an entire section?

Comment (dougszumski) [reply tasks/client.yml]: Does this help?  https://askubuntu.com/questions/659267/how-do-i-override-or-configure-systemd-services

Looks like  `systemctl edit foo` could be better than a template.

Comment (brtknr) [reply tasks/client.yml]: Hmm, not much fun to do with Ansible though... :( will have to resort to using shell: module...

Comment (markgoddard) [reply tasks/client.yml]: Here's the bit I forgot: you need to put it in /etc/systemd/system/foo.service.d/override.conf

File: files/beegfs-client.service.d/override.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Service]
```
Comment (markgoddard) [inline files/beegfs-client.service.d/override.conf]: A comment explaining the rationale for these changes will help future you and me.

**Comment by brtknr** [comment]
Relevant discussion on beefs-user group: https://groups.google.com/forum/#!topic/fhgfs-user/T99I3RR24KY

File: files/beegfs-client.service.d/override.conf (hunk 2 above)
Comment (brtknr) [reply files/beegfs-client.service.d/override.conf]: Acknowledged.

File: tasks/client.yml (hunk 1 above)
Comment (brtknr) [reply tasks/client.yml]: Done.

**Pull request merged**